File: search/linear_search.c
/* Linear search in C */

#include <stdio.h>
#include <stdlib.h>

#define N 1000

int main(void){
  int array[N], i, result_index;
  int y = 115;

  for(i = 0; i < N; i++) {
    array[i] = rand() % 1000 + 1;
  }

  printf("検索対象 = %d\n", y);

  for (i = 0; i < N; i++) {
    if (array[i] == y) {
      result_index = i;
      printf("%dはarrayの%d番目にあります \n", y, i + 1);

      return 0;
    }
  }

  printf("%dはarrayに存在しません \n", y);
  return 0;
}

File: search/linear_search.rb
# Linear search in Ruby

def find_number(array, y)
  result_index = nil
  array.each_with_index do |i, a|
    if a == y
      result_index = i
      break
    end
  end

  return p "#{y}はarrayの#{result_index + 1}番目にあります" if !result_index.nil?
  p "#{y}はarrayに存在しません"
end

find_number([*1..1000].shuffle, 115)

